main.py

```diff
@@ -37,14 +37,22 @@ def add_book():
     new_book.save()
 
 def delete_book():
-    print('Which book would you like to delete?/n')
-    book_id=ui.get_book_id()
-    book = store.get_book_by_id(book_id)
-    book.delete()
-    
-
+    print('Which book would you like to delete?')
+    while True:
+        try:
+            book_id=ui.get_book_id()
+            book = store.get_book_by_id(book_id)
+            if book is not None:
+                book.delete()
+                break
+            else:
+                print('\nThe book ID", book_id, " is NOT in the database\n')
+        except UnboundLocalError:
```

Comment:
This is not the right fix. The cause of the UnboundLocalError is an undefined variable somewhere else, in the original code the problem was here - if there is no book, an uninitialized book variable is returned which causes the UnboundLocalError. The fix is to fix the code in bookstore.py https://github.com/claraj/readinglist/blob/c4f1ce378ddd1df6e4303c93c42014133e04d31c/bookstore.py#L193-L198